### PR TITLE
Adsk contrib/add back channels attribute

### PR DESCRIPTION
**Pull request by feldstj**

### CODE REVIEW

File: documents/TestSuite/stdlib/channel/channels_attribute.mtlx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<!--
+
+Tests the channel attribute.
+
+-->
+<?xml version="1.0"?>
+<materialx version="1.36">
+  <nodegraph name="float_to_color4_out" type="">
```
Comment (bernardkwok) [inline documents/TestSuite/stdlib/channel/channels_attribute.mtlx]: For completeness can we add one test where the input is not a constant.
e.g. it's a stream or an image.

File: source/MaterialXCore/Interface.cpp [hunk 2]
```diff
@@ -64,13 +65,13 @@ bool PortElement::validate(string* message) const
             {
                 OutputPtr output = connectedNodeDef->getOutput(getOutputString());
                 validateRequire(output != nullptr, res, message, "Invalid output in port connection");
-                if (output)
+                if (output && !hasChannels())
```
Comment (bernardkwok) [inline source/MaterialXCore/Interface.cpp]: Should we check that at least the type is a vector/color or float. If it's another type channels is also not valid (e.g. matrix)

File: source/MaterialXGenOsl/OslShaderGenerator.cpp [hunk 3]
```diff
@@ -303,6 +303,10 @@ void OslShaderGenerator::emitFinalOutput(Shader& shader) const
     }
 
     string finalResult = outputSocket->connection->variable;
+    if (!outputSocket->channels.empty())
+    {
+        finalResult = _syntax->getSwizzledVariable(finalResult, outputSocket->connection->type, outputSocket->channels, outputSocket->type);
```
Comment (bernardkwok) [inline source/MaterialXGenOsl/OslShaderGenerator.cpp]: Is this always a valid possibility for final output? I thought that for OSL we cannot output to float, vec/color2 or vec4/color4 ? which could be possible with the swizzle. And for GLSL we can only allow vec3/vec4.

File: documents/TestSuite/stdlib/channel/channels_attribute.mtlx (hunk 1 above)
Comment (feldstj) [reply documents/TestSuite/stdlib/channel/channels_attribute.mtlx]: Added 2 image based tests.

File: source/MaterialXCore/Interface.cpp (hunk 2 above)
Comment (feldstj) [reply source/MaterialXCore/Interface.cpp]: Added a pretty extensive addition related to this.

File: source/MaterialXGenOsl/OslShaderGenerator.cpp (hunk 3 above)
Comment (feldstj) [reply source/MaterialXGenOsl/OslShaderGenerator.cpp]: OSL is allowed to output to any type now. For GLSL we only support vec4 but that is handled later, in the last section of this method.

**Pull request closed without merging**